src/mjolnir/graphenhancer.cc

```diff
@@ -1344,14 +1706,42 @@ void enhance(const boost::property_tree::ptree& pt,
           }
         }
 
+        // since the internal flag is set, we are at either the prior or the next edge and we need to
```

Comment:
For the next few lines of code, this is where we either leave or toss the lane information based on if the next edge is internal or not.